.github/workflows/test.yml

```
@@ -0,0 +1,33 @@
+name: Test
+
+on:
+  pull_request:
+    paths:
+      - '.github/workflows/test.yml'
+
+permissions: read-all
```

i think to giving too many permissions for CI execution

```suggestion
permissions:
  contents: read
```